src/main.cpp

```diff
@@ -132,6 +140,22 @@ int main(int argc, char** argv)
 
     MainWindow mainWindow;
 
+#ifndef QT_DEBUG
+    // Disable screen capture if capable and not explicitly allowed
+    if (osUtils->canPreventScreenCapture() && !parser.isSet(allowScreenCaptureOption)) {
+        // This ensures any top-level windows (Main Window, Modal Dialogs, etc.) are excluded from screenshots
+        QObject::connect(&app, &QGuiApplication::focusWindowChanged, &mainWindow, [&](QWindow* window) {
+            if (window) {
+                if (!osUtils->setPreventScreenCapture(window, true)) {
+                    mainWindow.displayGlobalMessage(
+                        QObject::tr("Warning: Failed to prevent screenshots on a top level window!"),
+                        MessageWidget::Error);
+                }
+            }
+        });
```

Comment:
This could be even more simplified by just hiding `MaingWindow` in each implementation (`getMainWindow()`). So you wouldn't need this connection being initialized here and `setPreventScreenCapture` can then just accept a single boolean parameter.
At least this is true on Windows, I haven't checked with macOS.

Reply:
I am unconcerned with file choice dialogs, those are actually owned by the OS and not KeePassXC process. We have to do more than mainwindow though to account for Auto-Type picker dialog, browser choice selection, totp dialogs, etc.

Reply:
You're right about this connection. I've missed the fact that `SetWindowDisplayAffinity` sets affinity flag only for the targeted window and not also for the windows that might be associated with it. Unfortunately yes, Qt doesn't provide any signal to notify you about new window being created. I've also tried to override `childEvent` && `event` for `Application` & `MainWindow` classes and set affinity flag to new child windows & widgets, but it turns out not all windows are processed through these event functions. So I guess this connection here has to stay.

For the system dialogs not being hidden, I've also forgot to mentioned that the app's menu widgets and all context widgets are also not hidden. I don't know if you noticed this. I'm not saying it's crucial for the app security but this does leak some information, like the name of the database, location where it is stored (i.e. cloud storage), does the entry has TOTP set etc... 
For these reasons and the benefit of better confidentiality, I would rather then go with more complex implementation i.e. per platform window handling.